app/domain/operations/hbx_enrollments/expire.rb

```diff
@@ -4,38 +4,193 @@ module Operations
+                                                                       transaction: request_transaction})
+        transaction_params      = yield construct_response_transaction_params
+        @response_transaction   = yield create_response_transaction(transaction_params,
+                                                                    { job: job,
+                                                                      transmission: request_transmission,
+                                                                      transaction: request_transaction})
+        result                   = yield expire_enrollment
 
         Success(result)
       end
 
       private
 
       def validate(params)
-        return Failure('Missing enrollment_hbx_id') unless params.key?(:enrollment_hbx_id)
-        Success(params[:enrollment_hbx_id])
+        unless params.is_a?(Hash)
+          msg = "Invalid input params: #{params}. Expected a hash."
+          return Failure(msg)
+        end
+
+        unless params[:transmittable_identifiers].is_a?(Hash)
+          msg = "Invalid transmittable_identifiers in params: #{params}. Expected a hash."
+          return Failure(msg)
+        end
+
+        if params[:enrollment_gid].blank?
+          msg = "Missing enrollment_gid in params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:job_gid].blank?
+          msg = "Missing job_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:transmission_gid].blank?
+          msg = "Missing transmission_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:transaction_gid].blank?
+          msg = "Missing transaction_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        if params[:transmittable_identifiers][:subject_gid].blank?
+          msg = "Missing subject_gid in transmittable_identifiers of params: #{params}."
+          return Failure(msg)
+        end
+
+        Success(params)
+      end
+
+      def find_enrollment(enrollment_gid)
+        hbx_enrollment = GlobalID::Locator.locate(enrollment_gid)
+
+        if hbx_enrollment.blank?
+          msg = "No HbxEnrollment found with given global ID: #{enrollment_gid}"
+          Failure(msg)
+        elsif !hbx_enrollment.is_ivl_by_kind?
+          msg = "Failed to expire enrollment hbx id #{hbx_enrollment.hbx_id} - #{hbx_enrollment.kind} is not a valid IVL enrollment kind"
+          Failure(msg)
+        else
+          Success(hbx_enrollment)
+        end
       end
 
-      def find_enrollment(enrollment_hbx_id)
-        Operations::HbxEnrollments::Find.new.call({hbx_id: enrollment_hbx_id})
+      def find_and_update_request_transmission(transmission_gid, job)
```

Comment:
I think we should simply return a failure monad if we are unable to find the Transmission. This operation can be called with invalid params and we should not add errors to the job. This specific step should be split into 2 steps.
- Find the transmission by global_id
- Updating the Status of Transmission

Reply:
I agree with @saikumar9.  If we are checking for success/failure of finding the transmission inside this step, it seems more appropriate to move that into its own step in the `call` method up top.  Same goes for the `find_and_update_request_transaction` method and anywhere else we are checking for success/failure inside a method that returns a separate success/failure.